File: kinotic-domain/src/main/java/org/kinotic/domain/api/model/OrganizationStorageStatus.java
package org.kinotic.domain.api.model;

/**
 * Lifecycle state of an organization's storage, and why when something is wrong.
 *
 * @param type    the lifecycle state of the storage
 * @param message why the storage is in this state, or null when the state speaks for
 *                itself; typically the failure that left it
 *                {@link OrganizationStorageStatusType#FAILED}
 */
public record OrganizationStorageStatus(OrganizationStorageStatusType type, String message) {

    public OrganizationStorageStatus(OrganizationStorageStatusType type) {
        this(type, null);
    }
}

File: kinotic-js/workspace/packages/management-api/src/api/model/OrganizationStorageStatus.ts
import { OrganizationStorageStatusType } from '@/api/model/OrganizationStorageStatusType'

/**
 * Lifecycle state of an organization's storage, and why when something is wrong.
 */
export interface OrganizationStorageStatus {
    /**
     * The lifecycle state of the storage.
     */
    type: OrganizationStorageStatusType
    /**
     * Why the storage is in this state, or null when the state speaks for itself;
     * typically the failure that left it FAILED.
     */
    message: string | null
}
